From 1610bd2ce98e3d93296fd3bc4d2c24e905428a4a Mon Sep 17 00:00:00 2001
From: Juan Quintela <quintela@redhat.com>
Date: Tue, 3 Aug 2021 11:24:26 +0200
Subject: [PATCH 06/39] Remove RHEL 7.3.0 machine types

RH-Author: quintela1 <quintela@redhat.com>
RH-MergeRequest: 30: Remove RHEL 7.x.0 machine types
RH-Commit: [4/6] 7a674496dd29a7a87843770fd0459b85831cc866 (juan.quintela/qemu-kvm)
RH-Bugzilla: 1968519
RH-Acked-by: Cornelia Huck <cohuck@redhat.com>
RH-Acked-by: Thomas Huth <thuth@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>

Only thing to remove with care was __redhat_e1000e_7_3_intr_state.

Signed-off-by: Juan Quintela <quintela@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/core/machine.c    | 22 ---------------------
 hw/i386/pc.c         | 28 ---------------------------
 hw/i386/pc_piix.c    | 19 ------------------
 hw/i386/pc_q35.c     | 19 ------------------
 hw/net/e1000e.c      | 22 ---------------------
 hw/ppc/spapr.c       | 46 --------------------------------------------
 include/hw/boards.h  |  3 ---
 include/hw/i386/pc.h |  3 ---
 8 files changed, 162 deletions(-)

diff --git a/hw/core/machine.c b/hw/core/machine.c
index 54eb8376a7..0bcaabd8a9 100644
--- a/hw/core/machine.c
+++ b/hw/core/machine.c
@@ -224,28 +224,6 @@ GlobalProperty hw_compat_rhel_7_4[] = {
 };
 
 const size_t hw_compat_rhel_7_4_len = G_N_ELEMENTS(hw_compat_rhel_7_4);
-/* Mostly like HW_COMPAT_2_6 + HW_COMPAT_2_7 + HW_COMPAT_2_8 except
- * disable-modern, disable-legacy, page-per-vq have already been
- * backported to RHEL7.3
- */
-GlobalProperty hw_compat_rhel_7_3[] = {
-    { "virtio-mmio", "format_transport_address", "off" },
-    { "virtio-serial-device", "emergency-write", "off" },
-    { "ioapic", "version", "0x11" },
-    { "intel-iommu", "x-buggy-eim", "true" },
-    { "virtio-pci", "x-ignore-backend-features", "on" },
-    { "fw_cfg_mem", "x-file-slots", stringify(0x10) },
-    { "fw_cfg_io", "x-file-slots", stringify(0x10) },
-    { "pflash_cfi01", "old-multiple-chip-handling", "on" },
-    { TYPE_PCI_DEVICE, "x-pcie-extcap-init", "off" },
-    { "virtio-pci", "x-pcie-deverr-init", "off" },
-    { "virtio-pci", "x-pcie-lnkctl-init", "off" },
-    { "virtio-pci", "x-pcie-pm-init", "off" },
-    { "virtio-net-device", "x-mtu-bypass-backend", "off" },
-    { "e1000e", "__redhat_e1000e_7_3_intr_state", "on" },
-};
-const size_t hw_compat_rhel_7_3_len = G_N_ELEMENTS(hw_compat_rhel_7_3);
-
 GlobalProperty hw_compat_5_2[] = {
     { "ICH9-LPC", "smm-compat", "on"},
     { "PIIX4_PM", "smm-compat", "on"},
diff --git a/hw/i386/pc.c b/hw/i386/pc.c
index 938cb82818..75abe0acc2 100644
--- a/hw/i386/pc.c
+++ b/hw/i386/pc.c
@@ -491,34 +491,6 @@ GlobalProperty pc_rhel_7_4_compat[] = {
 };
 const size_t pc_rhel_7_4_compat_len = G_N_ELEMENTS(pc_rhel_7_4_compat);
 
-GlobalProperty pc_rhel_7_3_compat[] = {
-    /* pc_rhel_7_3_compat from pc_compat_2_8 */ 
-    { "kvmclock", "x-mach-use-reliable-get-clock", "off" },
-    /* pc_rhel_7_3_compat from pc_compat_2_7 */ 
-    { TYPE_X86_CPU, "l3-cache", "off" },
-    /* pc_rhel_7_3_compat from pc_compat_2_7 */ 
-    { TYPE_X86_CPU, "full-cpuid-auto-level", "off" },
-    /* pc_rhel_7_3_compat from pc_compat_2_7 */ 
-    { "Opteron_G3" "-" TYPE_X86_CPU, "family", "15" },
-    /* pc_rhel_7_3_compat from pc_compat_2_7 */ 
-    { "Opteron_G3" "-" TYPE_X86_CPU, "model", "6" },
-    /* pc_rhel_7_3_compat from pc_compat_2_7 */ 
-    { "Opteron_G3" "-" TYPE_X86_CPU, "stepping", "1" },
-    /* pc_rhel_7_3_compat from pc_compat_2_7 */ 
-    { "isa-pcspk", "migrate", "off" },
-    /* pc_rhel_7_3_compat from pc_compat_2_6 */ 
-    { TYPE_X86_CPU, "cpuid-0xb", "off" },
-    /* pc_rhel_7_3_compat from pc_compat_2_8 */ 
-    { "ICH9-LPC", "x-smi-broadcast", "off" },
-    /* pc_rhel_7_3_compat from pc_compat_2_8 */ 
-    { TYPE_X86_CPU, "vmware-cpuid-freq", "off" },
-    /* pc_rhel_7_3_compat from pc_compat_2_8 */ 
-    { "Haswell-" TYPE_X86_CPU, "stepping", "1" },
-    /* pc_rhel_7_3_compat from pc_compat_2_3 added in 2.9*/ 
-    { TYPE_X86_CPU, "kvm-no-smi-migration", "on" },
-};
-const size_t pc_rhel_7_3_compat_len = G_N_ELEMENTS(pc_rhel_7_3_compat);
-
 /*
  * The PC_RHEL_*_COMPAT serve the same purpose for RHEL-7 machine
  * types as the PC_COMPAT_* do for upstream types.
diff --git a/hw/i386/pc_piix.c b/hw/i386/pc_piix.c
index 201cbbdb01..64662cc3d5 100644
--- a/hw/i386/pc_piix.c
+++ b/hw/i386/pc_piix.c
@@ -1030,22 +1030,3 @@ static void pc_machine_rhel740_options(MachineClass *m)
 
 DEFINE_PC_MACHINE(rhel740, "pc-i440fx-rhel7.4.0", pc_init_rhel740,
                   pc_machine_rhel740_options);
-
-static void pc_init_rhel730(MachineState *machine)
-{
-    pc_init1(machine, TYPE_I440FX_PCI_HOST_BRIDGE, \
-             TYPE_I440FX_PCI_DEVICE);
-}
-
-static void pc_machine_rhel730_options(MachineClass *m)
-{
-    PCMachineClass *pcmc = PC_MACHINE_CLASS(m);
-    pc_machine_rhel740_options(m);
-    m->desc = "RHEL 7.3.0 PC (i440FX + PIIX, 1996)";
-    pcmc->linuxboot_dma_enabled = false;
-    compat_props_add(m->compat_props, hw_compat_rhel_7_3, hw_compat_rhel_7_3_len);
-    compat_props_add(m->compat_props, pc_rhel_7_3_compat, pc_rhel_7_3_compat_len);
-}
-
-DEFINE_PC_MACHINE(rhel730, "pc-i440fx-rhel7.3.0", pc_init_rhel730,
-                  pc_machine_rhel730_options);
diff --git a/hw/i386/pc_q35.c b/hw/i386/pc_q35.c
index 01ff3e0544..bf49a943dc 100644
--- a/hw/i386/pc_q35.c
+++ b/hw/i386/pc_q35.c
@@ -792,22 +792,3 @@ static void pc_q35_machine_rhel740_options(MachineClass *m)
 
 DEFINE_PC_MACHINE(q35_rhel740, "pc-q35-rhel7.4.0", pc_q35_init_rhel740,
                   pc_q35_machine_rhel740_options);
-
-static void pc_q35_init_rhel730(MachineState *machine)
-{
-    pc_q35_init(machine);
-}
-
-static void pc_q35_machine_rhel730_options(MachineClass *m)
-{
-    PCMachineClass *pcmc = PC_MACHINE_CLASS(m);
-    pc_q35_machine_rhel740_options(m);
-    m->desc = "RHEL-7.3.0 PC (Q35 + ICH9, 2009)";
-    m->max_cpus = 255;
-    pcmc->linuxboot_dma_enabled = false;
-    compat_props_add(m->compat_props, hw_compat_rhel_7_3, hw_compat_rhel_7_3_len);
-    compat_props_add(m->compat_props, pc_rhel_7_3_compat, pc_rhel_7_3_compat_len);
-}
-
-DEFINE_PC_MACHINE(q35_rhel730, "pc-q35-rhel7.3.0", pc_q35_init_rhel730,
-                  pc_q35_machine_rhel730_options);
diff --git a/hw/net/e1000e.c b/hw/net/e1000e.c
index 6d39c1f1c4..b22f232863 100644
--- a/hw/net/e1000e.c
+++ b/hw/net/e1000e.c
@@ -79,12 +79,6 @@ struct E1000EState {
     bool disable_vnet;
 
     E1000ECore core;
-
-    /* 7.3 had the intr_state field that was in the original e1000e code
-     * but that was removed prior to 2.7's release
-     */
-    bool redhat_7_3_intr_state_enable;
-    uint32_t redhat_7_3_intr_state;
 };
 
 #define E1000E_MMIO_IDX     0
@@ -100,10 +94,6 @@ struct E1000EState {
 #define E1000E_MSIX_TABLE   (0x0000)
 #define E1000E_MSIX_PBA     (0x2000)
 
-/* Values as in RHEL 7.3 build and original upstream */
-#define RH_E1000E_USE_MSI     BIT(0)
-#define RH_E1000E_USE_MSIX    BIT(1)
-
 static uint64_t
 e1000e_mmio_read(void *opaque, hwaddr addr, unsigned size)
 {
@@ -315,8 +305,6 @@ e1000e_init_msix(E1000EState *s)
     } else {
         if (!e1000e_use_msix_vectors(s, E1000E_MSIX_VEC_NUM)) {
             msix_uninit(d, &s->msix, &s->msix);
-        } else {
-            s->redhat_7_3_intr_state |= RH_E1000E_USE_MSIX;
         }
     }
 }
@@ -488,8 +476,6 @@ static void e1000e_pci_realize(PCIDevice *pci_dev, Error **errp)
     ret = msi_init(PCI_DEVICE(s), 0xD0, 1, true, false, NULL);
     if (ret) {
         trace_e1000e_msi_init_fail(ret);
-    } else {
-        s->redhat_7_3_intr_state |= RH_E1000E_USE_MSI;
     }
 
     if (e1000e_add_pm_capability(pci_dev, e1000e_pmrb_offset,
@@ -613,11 +599,6 @@ static const VMStateDescription e1000e_vmstate_intr_timer = {
     VMSTATE_STRUCT_ARRAY(_f, _s, _num, 0,                           \
                          e1000e_vmstate_intr_timer, E1000IntrDelayTimer)
 
-static bool rhel_7_3_check(void *opaque, int version_id)
-{
-    return ((E1000EState *)opaque)->redhat_7_3_intr_state_enable;
-}
-
 static const VMStateDescription e1000e_vmstate = {
     .name = "e1000e",
     .version_id = 1,
@@ -629,7 +610,6 @@ static const VMStateDescription e1000e_vmstate = {
         VMSTATE_MSIX(parent_obj, E1000EState),
 
         VMSTATE_UINT32(ioaddr, E1000EState),
-        VMSTATE_UINT32_TEST(redhat_7_3_intr_state, E1000EState, rhel_7_3_check),
         VMSTATE_UINT32(core.rxbuf_min_shift, E1000EState),
         VMSTATE_UINT8(core.rx_desc_len, E1000EState),
         VMSTATE_UINT32_ARRAY(core.rxbuf_sizes, E1000EState,
@@ -678,8 +658,6 @@ static PropertyInfo e1000e_prop_disable_vnet,
 
 static Property e1000e_properties[] = {
     DEFINE_NIC_PROPERTIES(E1000EState, conf),
-    DEFINE_PROP_BOOL("__redhat_e1000e_7_3_intr_state", E1000EState,
-                        redhat_7_3_intr_state_enable, false),
     DEFINE_PROP_SIGNED("disable_vnet_hdr", E1000EState, disable_vnet, false,
                         e1000e_prop_disable_vnet, bool),
     DEFINE_PROP_SIGNED("subsys_ven", E1000EState, subsys_ven,
diff --git a/hw/ppc/spapr.c b/hw/ppc/spapr.c
index 0bc558e0b7..ca0b99403e 100644
--- a/hw/ppc/spapr.c
+++ b/hw/ppc/spapr.c
@@ -5351,52 +5351,6 @@ static void spapr_machine_rhel740sxxm_class_options(MachineClass *mc)
 
 DEFINE_SPAPR_MACHINE(rhel740sxxm, "rhel7.4.0-sxxm", false);
 
-/*
- * pseries-rhel7.3.0
- * like spapr_compat_2_6/_2_7/_2_8 but "ddw" has been backported to RHEL7_3
- */
-GlobalProperty spapr_compat_rhel7_3[] = {
-    { TYPE_SPAPR_PCI_HOST_BRIDGE, "mem_win_size", "0xf80000000" },
-    { TYPE_SPAPR_PCI_HOST_BRIDGE, "mem64_win_size", "0" },
-    { TYPE_POWERPC_CPU, "pre-2.8-migration", "on" },
-    { TYPE_SPAPR_PCI_HOST_BRIDGE, "pre-2.8-migration", "on" },
-    { TYPE_SPAPR_PCI_HOST_BRIDGE, "pcie-extended-configuration-space", "off" },
-};
-const size_t spapr_compat_rhel7_3_len = G_N_ELEMENTS(spapr_compat_rhel7_3);
-
-static void spapr_machine_rhel730_class_options(MachineClass *mc)
-{
-    SpaprMachineClass *smc = SPAPR_MACHINE_CLASS(mc);
-
-    spapr_machine_rhel740_class_options(mc);
-    mc->default_cpu_type = POWERPC_CPU_TYPE_NAME("power7_v2.3");
-    mc->default_machine_opts = "modern-hotplug-events=off";
-    compat_props_add(mc->compat_props, hw_compat_rhel_7_3, hw_compat_rhel_7_3_len);
-    compat_props_add(mc->compat_props, spapr_compat_rhel7_3, spapr_compat_rhel7_3_len);
-
-    smc->phb_placement = phb_placement_2_7;
-}
-
-DEFINE_SPAPR_MACHINE(rhel730, "rhel7.3.0", false);
-
-/*
- * pseries-rhel7.3.0-sxxm
- *
- * pseries-rhel7.3.0 with speculative execution exploit mitigations enabled by default
- */
-
-static void spapr_machine_rhel730sxxm_class_options(MachineClass *mc)
-{
-    SpaprMachineClass *smc = SPAPR_MACHINE_CLASS(mc);
-
-    spapr_machine_rhel730_class_options(mc);
-    smc->default_caps.caps[SPAPR_CAP_CFPC] = SPAPR_CAP_WORKAROUND;
-    smc->default_caps.caps[SPAPR_CAP_SBBC] = SPAPR_CAP_WORKAROUND;
-    smc->default_caps.caps[SPAPR_CAP_IBS] = SPAPR_CAP_FIXED_CCD;
-}
-
-DEFINE_SPAPR_MACHINE(rhel730sxxm, "rhel7.3.0-sxxm", false);
-
 static void spapr_machine_register_types(void)
 {
     type_register_static(&spapr_machine_info);
diff --git a/include/hw/boards.h b/include/hw/boards.h
index 43eb868ceb..24d4d726d8 100644
--- a/include/hw/boards.h
+++ b/include/hw/boards.h
@@ -439,7 +439,4 @@ extern const size_t hw_compat_rhel_7_5_len;
 extern GlobalProperty hw_compat_rhel_7_4[];
 extern const size_t hw_compat_rhel_7_4_len;
 
-extern GlobalProperty hw_compat_rhel_7_3[];
-extern const size_t hw_compat_rhel_7_3_len;
-
 #endif
diff --git a/include/hw/i386/pc.h b/include/hw/i386/pc.h
index c26c6dcc72..820fb3f577 100644
--- a/include/hw/i386/pc.h
+++ b/include/hw/i386/pc.h
@@ -305,9 +305,6 @@ extern const size_t pc_rhel_7_5_compat_len;
 extern GlobalProperty pc_rhel_7_4_compat[];
 extern const size_t pc_rhel_7_4_compat_len;
 
-extern GlobalProperty pc_rhel_7_3_compat[];
-extern const size_t pc_rhel_7_3_compat_len;
-
 /* Helper for setting model-id for CPU models that changed model-id
  * depending on QEMU versions up to QEMU 2.4.
  */
-- 
2.27.0

